packages/terra-pills/src/FilterPills.jsx

```diff
@@ -0,0 +1,448 @@
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleOnRemove = (pillKey, metaData, event) => {
+    if (event.type === 'click') {
+      const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+      const targetId = event.target.previousSibling.getAttribute('id');
+      const currentIndex = pills.findIndex((element) => element.id === targetId);
+
+      if (pills.length > 1) {
+        setTabIndex('-1');
+        if (currentIndex === 0) {
+          focusNode.current = 0;
+        } else {
+          focusNode.current = currentIndex - 1;
+        }
+        setTabIndex('0');
+      }
+    }
+
+    if (onRemove) {
+      onRemove(pillKey, metaData);
+    }
+  };
+
+  // set the focus to current pill if the pill is clicked with mouse
+  const handleOnPillSelect = (pillRef, pillKey, metaData, event) => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    const targetId = event.target.getAttribute('id');
+
+    if (targetId && event.target.hasAttribute('data-terra-pill')) {
+      setTabIndex('-1');
+      currentPill.current = targetId;
+      focusNode.current = pills.findIndex((element) => element.id === targetId);
+      setTabIndex('0');
+      focusCurrentNode();
+    }
+  };
+
+  const handleOnSelectRollUp = (event) => {
+    const pills = [...filterPillsRef.current.querySelectorAll('[data-terra-pill]')];
+    if (isCollapsible && isCollapsed) {
+      if (event.type === 'keydown') {
+        isRollUpRemoved.current = true;
+        focusNode.current = pills.length;
+      } else {
+        setTabIndex('-1');
+        focusNode.current = pills.length;
+      }
+    } else {
+      setShowRollupPillInteraction(false);
+    }
+    setIsCollapsed(!isCollapsed);
+  };
+
+  const handlePillListOnblur = () => setContainerTabindex('-1');
+
+  const filterPillsProps = {};
+  filterPillsProps.onKeyDown = handlePillListKeyDown;
+  filterPillsProps.onBlur = handlePillListOnblur;
+
+  const pillListClassNames = classNames(
+    cx([
+      'pills-group',
+      theme.className,
+    ]),
+    customProps.className,
+  );
+
+  const pillInteractionHintID = 'terra-pills-interaction-hint';
+  const pillGroupInteractionHintID = 'terra-pills-group-interaction-hint';
+  let removedPillInteractionHint = '';
+  const pillGroupAriaDescribedBy = ariaDescribedBy ? `${ariaDescribedBy} ${pillGroupInteractionHintID}` : pillGroupInteractionHintID;
+  let pillGroupInteractionHint = intl.formatMessage({ id: 'Terra.pills.hint.pillList' }, { numberOfPills: React.Children.count(children) });
+  if (isCollapsible && (rollUpCount > 0) && isCollapsed) {
+    pillGroupInteractionHint += `, ${intl.formatMessage({ id: 'Terra.pills.hint.rollupNotVisible' }, { pillsNotVisibleCount: rollUpCount })}`;
+  } else if (isCollapsible && !isCollapsed && showRollupPillInteraction) {
+    removedPillInteractionHint = 'hidden items are displayed now...'; // needs transtaions lp052179
```

Comment:
Are we still waiting for translations for these strings or is this the translated string that can now be used?

`"Terra.pills.hint.rollupVisible": "{pillsNotVisibleCount} hidden items are now displayed"`

Reply:
Changed cfb6aad5b38a9957f1449aa2f75bbf5d001c3c96